FreeRTOS/Test/CMock/stream_buffer/stream_buffer_utest.c

```diff
@@ -455,40 +488,44 @@ void test_xStreamBufferCreate_malloc_fail( void )
 }
 
 /**
- * @brief Assertion fails if a zero buffer size is passed as  the parameter.
+ * @brief Assertion fails if a zero buffer size is passed as the xBufferSizeBytes parameter.
  */
 void test_xStreamBufferCreate_zero_buffer_size( void )
 {
+    xStreamBuffer = xStreamBufferCreate( 0, TEST_STREAM_BUFFER_TRIGGER_LEVEL );
 
-    if( TEST_PROTECT() )
-    {
-        ( void ) xStreamBufferCreate( 0, TEST_STREAM_BUFFER_TRIGGER_LEVEL );
-    }
-    validate_and_clear_assertions();
+    validate_and_clear_multiple_assertions();
+
+    TEST_ASSERT_EQUAL( 1U + sizeof( StaticStreamBuffer_t ), dynamicMemoryAllocated );
+
+    TEST_ASSERT_NOT_NULL( xStreamBuffer );
+    vStreamBufferDelete( xStreamBuffer );
 }
 
 /**
  * @brief Assertion fails if trigger level is greater than the stream buffer size.
  */
 void test_xStreamBufferCreate_invalid_trigger_level( void )
 {
-    if( TEST_PROTECT() )
-    {
-        ( void ) xStreamBufferCreate( TEST_STREAM_BUFFER_SIZE, ( TEST_STREAM_BUFFER_SIZE + 1 ) );
-    }
-    validate_and_clear_assertions();
+    xStreamBuffer = xStreamBufferCreate( TEST_STREAM_BUFFER_SIZE, ( TEST_STREAM_BUFFER_SIZE + 1 ) );
```

Comment:
Why is the assertion turned off and we continue execution here.

Reply:
Asserts can be turned off in production. This tests both the configASSERT disabled and configASSERT enabled cases.